01_fib/saurabhvyas9/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Abs returns the absolute value of x.
+func Abs(x int) int {
+	if x < 0 {
+		return -x
+	}
+	return x
+}
+
+// Returns Fibonacci and Negafibonacci series
+func fib(n int) {
+	m := Abs(n)
+	a, b, neg := 0, 1, 1
+	for i := 1; i <= m; i++ {
+		if (i%2) == 0 && n < 0 {
+			neg = -1
+		}
+		fmt.Fprintln(out, neg*b)
+		a, b = b, a+b
+		neg = 1
```

Comment:
I would have moved this assignment to the top of the loop. Infect I think you could have also defined it there (`neg := 1`).

Reply:
Done.